### PR TITLE
Adds DPanic and Panic level logging (writes log, then panics)

### PR DESCRIPTION
**Pull request by paulnpdev**
**What changed?**
New methods on logger to allow panics to create consistent logger entries.

**Why?**
Tired of panic strings that don't have enough information.

**How did you test it?**
New tests on logger.

**Potential risks**
Since no code is yet using these new methods, risk should be minimal.

**Is hotfix candidate?**
No.

### CODE REVIEW

File: common/log/zap_logger.go [hunk 1]
```diff
@@ -197,15 +214,17 @@ func buildZapLogger(cfg Config, disableCaller bool) *zap.Logger {
...
 	}
-
+	development := false // default to Prod
+	if cfg.DevelopmentStage == true {
+		development = true
+	}
 	config := zap.Config{
 		Level:            zap.NewAtomicLevelAt(parseZapLevel(cfg.Level)),
-		Development:      false,
+		Development:      development,
```
Comment (yux0) [inline common/log/zap_logger.go:227]: cfg.DevelopmentStage？

**Review by alexshtin — APPROVED** [review]
> Tired of panic strings that don't have enough information.

you mean, log strings, that don't have correct call stack and with `DPanic` we will get the right call stack to the actual error.

File: common/log/interface.go [hunk 2]
```diff
@@ -44,6 +44,8 @@ type (
 		Info(msg string, tags ...tag.Tag)
 		Warn(msg string, tags ...tag.Tag)
 		Error(msg string, tags ...tag.Tag)
+		DPanic(msg string, tags ...tag.Tag)
+		Panic(msg string, tags ...tag.Tag)
```
Comment (alexshtin) [inline common/log/interface.go:48]: I have to point out that this is a breaking change for anyone who has custom logger and provide it using `serverOptions`. I believe we can afford it but need to mention  on 1.20 release notes. I am tagging this PR with 1.20 tag.

**Comment by paulnpdev** [comment]
> > Tired of panic strings that don't have enough information.
> 
> you mean, log strings, that don't have correct call stack and with `DPanic` we will get the right call stack to the actual error.

both

File: common/log/interface.go (hunk 2 above)
Comment (paulnpdev) [reply common/log/interface.go:48]: fortunately it will break at compile time

**Pull request merged**